Visualizer/src/SimulationObjects/Simulation.js

```diff
@@ -53,7 +70,15 @@ export default class Simulation {
     this.agents = this.owners
       .map((owner) => owner.agents)
       .flat()
-      .sort((a, b) => (first(Object.keys(a.combinedPath.ticks)) < first(Object.keys(b.combinedPath.ticks)) ? -1 : 1));
+      .sort((a, b) => {
+        if (!a.combinedPath) {
+          return -1;
+        }
+        if (!b.combinedPath) {
+          return 1;
+        }
+        return first(Object.keys(a.combinedPath.ticks)) < first(Object.keys(b.combinedPath.ticks)) ? -1 : 1;
+      });
```

Comment:
this is a hacky fix to prevent the "combinedPath is undefined" error, which still happens sometimes